test/replication-luatest/gh_8433_replication_timeout_test.lua

```diff
@@ -0,0 +1,87 @@
+local luatest = require('luatest')
+local server = require('luatest.server')
+local replica_set = require('luatest.replica_set')
+
+local g = luatest.group('prevote_fail')
+
+local function get_term(server)
+    return server:exec(function()
+        return box.info.election.term
+    end)
+end
+
+g.after_all(function(g)
+    g.replica_set:stop()
+end)
+
+g.before_all(function(g)
+    g.replica_set = replica_set:new({})
+    g.box_cfg = {
+        election_mode = 'candidate',
+        replication = {
+            server.build_listen_uri('server1'),
+            server.build_listen_uri('server2'),
+            server.build_listen_uri('server3'),
+        },
+    }
+
+    g.server1 = g.replica_set:build_and_add_server{
+        alias = 'server1',
+        box_cfg = g.box_cfg,
+    }
+
+    g.server2 = g.replica_set:build_and_add_server{
+        alias = 'server2',
+        box_cfg = g.box_cfg,
+    }
+
+    g.server3 = g.replica_set:build_and_add_server{
+        alias = 'server3',
+        box_cfg = g.box_cfg,
+    }
+
+    g.replica_set:start()
+    g.replica_set:wait_for_fullmesh()
+
+    for _, server in ipairs(g.replica_set.servers) do
+        server:exec(function()
+            box.error.injection.set('ERRINJ_RAFT_BYPASS_CB_ELECTION_START', 1)
+        end)
+    end
+end)
+
+g.test_prevote_fail = function(g)
+    --
+    -- The test is flaky if not launched on debug version. The following
+    -- situation is acceptable and must be bypassed with error injection
+    -- (ERRINJ_RAFT_BYPASS_CB_ELECTION_START - forbids to start election
```

Comment:
Since you already added an injection to prevent servers from starting elections, I propose to lower replication timeout:
```diff
diff --git a/test/replication-luatest/gh_8433_replication_timeout_test.lua b/test/replication-luatest/gh_8433_replication_timeout_test.lua
index 959915cb1..a4d874234 100644
--- a/test/replication-luatest/gh_8433_replication_timeout_test.lua
+++ b/test/replication-luatest/gh_8433_replication_timeout_test.lua
@@ -23,6 +23,7 @@ g.before_all(function(g)
             server.build_listen_uri('server2'),
             server.build_listen_uri('server3'),
         },
+        replication_timeout = 0.1,
     }
 
     g.server1 = g.replica_set:build_and_add_server{
@@ -69,7 +70,7 @@ g.test_prevote_fail = function(g)
 
     local leader = g.replica_set:get_leader()
     leader:exec(function()
-        box.cfg({replication_timeout = 0.1})
+        box.cfg({replication_timeout = 0.01})
     end)
 
     local term = 0
```

With this change the test runs for ~ 2.5 seconds instead of ~ 7.

Reply:
I've rewritten test fully, no additional injections are required now. But we can still use timeout of 0.01, which works much faster

Reply:
Good job with stabilizing the test!